docs/guides/tutorials/chatgpt_bot.rst

```diff
@@ -0,0 +1,2316 @@
+          "autoprefixer": "10.4.14",
+          "eslint": "8.46.0",
+          "eslint-config-next": "13.4.13",
+          "postcss": "8.4.27",
+          "tailwindcss": "3.3.3"
+        }
+      }
+
+Now we can invoke the ``generate-embeddings.ts`` script from our terminal using
+a simple command:
+
+.. code-block:: bash
+
+   $ npm run embeddings
+
+After the script finishes, open the EdgeDB UI.
+
+.. code-block:: bash
+
+  $ edgedb ui
+
+Open your "edgedb" database and switch to the Data Explorer tab. You should see
+that the database has been updated with the embeddings and other relevant data.
+
+
+Answering user questions
+========================
+
+Now that we have the content's embeddings stored, we can start working on the
+handler for user questions. The user will submit a question to our server, and
+the handler will send them an answer back. We will define a route and an HTTP
+request handler for this task. Thanks to the power of Next.js, we can do all of
+this within our project using a `route handler`_.
+
+.. _route handler:
+  https://nextjs.org/docs/app/building-your-application/routing/route-handlers
+
+As we write our handler, one important consideration is that answers can be
+quite long. We could wait on the server side to get the whole answer from
+OpenAI and then send it to the client, but that would feel slow to the user.
+OpenAI supports streaming, so instead we can send the answer to the client in
+chunks, as they arrive to the server. With this approach, the user doesn't have
+to wait for the entire response before they start getting feedback and our API
+seems faster.
+
+In order to stream responses, we will use the browser's `server-sent events
+(SSE) API`_. Server-sent events enables a client to receive automatic updates
+from a server via an HTTP connection, and describes how servers can initiate
+data transmissions to clients once an initial client connection has been
+established. The client sends a request and with that request initiates a
+connection with our server. With that connection established, the server will
+send data back to the client in chunks until all of the data is sent, at which
+point it closes the connection.
+
+.. lint-off
+
+.. _server-sent events (SSE) API:
+  https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events
+
+.. lint-on
+
+
+Next.js route handler
+---------------------
+
+When using `Next.js's App Router <https://nextjs.org/docs/app>`_, route
+handlers should be written inside an ``app/api`` folder. Every route should
+have its own folder within that, and the handlers should be defined inside a
+``route.ts`` file inside the route's folder.
+
+Let's create a new folder for the answer generation route inside ``app/api``.
+
+.. code-block:: bash
+
+    $ mkdir app/api && cd app/api
+    $ mkdir generate-answer && touch generate-answer/route.ts
+
+As the final setup step, we will install the ``common-tags`` NPM package which
```

Comment:
We are not really generating HTML, codeBlock and oneLineTrim are more used to properly indent/unindent code:  

We also need to install the ``common-tags`` NPM package (and its corresponding
types package) which gives us some useful template tags that  we will use later
when we create the prompt from user's question and related sections.

.. code-block:: bash

    $ npm install common-tags @types/common-tags

Reply:
I updated in the example repo to use stripIndents instead of codeBlock.

Reply:
That's also from `common-tags`? I can't see the change on the examples repo yet.